rust/connlib/tunnel/src/tests/reference.rs

```diff
@@ -132,9 +136,22 @@ impl ReferenceStateMachine for ReferenceState {
             )
             .with(
                 1,
-                (cidr_resource(8), sample::select(state.all_gateways())).prop_map(
-                    |(resource, gateway)| Transition::AddCidrResource { resource, gateway },
-                ),
+                (cidr_resource(8), sample::select(state.all_gateways()))
```

Comment:
I think at this complexity level, we may want to extract a dedicated strategy function now. Also allows us to make a comment explaining this.